ab_host-api/storage.adoc

```diff
@@ -3,11 +3,46 @@
 
 Interface for accessing the storage from within the runtime.
 
-IMPORTANT: As of now, the storage API should silently ignore any keys that start
-with the `:child_storage:default:` prefix. This applies to reading and writing.
-If the function expects a return value, then _None_ (<<defn-option-type>>)
-should be returned. See
-https://github.com/paritytech/substrate/issues/12461[substrate issue #12461].
+[#sect-storage-assumptions]
+==== Implementation Assumptions
+The storage and child storage (<<sect-child-storage-api>>) API in the Substrate
+implementation makes some behavioral assumptions on the underlying storage
+architecture. While Polkadot Host implementers can decide for themselves on how
+those APIs are implemented, those behaviors must be replicated in order for the
+Runtime to be executed deterministically. In Substrate, child storages are
+namespaced respectively segregated by attaching the `:child_storage:default:`
+prefix to every child storage keys, followed by `:<KEY>` for the key within that
+child storage. However, the storage API described in this section and the child
+storage API share the same database. This means that the storage API can
+retrieve child storage entries.
+
+For example, calling `ext_storage_get_version_1` (<<sect-ext-storage-get>>) with
```

Comment:
Why is this not explained under `ext_storage_get_version_1`?
And why "for example"? We want a precise list, not examples.